python/ray/_private/runtime_env/mpi.py

```diff
@@ -0,0 +1,76 @@
+import sys
+import logging
+import argparse
+from typing import List, Optional
+from ray._private.runtime_env.context import RuntimeEnvContext
+from ray._private.runtime_env.plugin import RuntimeEnvPlugin
+import subprocess
+
+default_logger = logging.getLogger(__name__)
+
+
+class MPIPlugin(RuntimeEnvPlugin):
+    priority = 90
+    name = "mpi"
+
+    def modify_context(
+        self,
+        uris: List[str],  # noqa: ARG002
+        runtime_env: "RuntimeEnv",  # noqa: F821 ARG002
+        context: RuntimeEnvContext,
+        logger: Optional[logging.Logger] = default_logger,  # noqa: ARG002
+    ) -> None:
+        mpi_config = runtime_env.mpi()
+        if mpi_config is None:
+            return
+        try:
+            proc = subprocess.run(
+                ["mpirun", "--version"], capture_output=True, check=True
+            )
+        except subprocess.CalledProcessError:
+            logger.error(
+                "Failed to run mpi run. Please make sure mpi has been installed"
+            )
```

Comment:
should we kill proc here? Or does it guarantee the proc is killed? (if so can you comment here?)

Reply:
Modify context is in runtime env agent I thin. Exception should be good? I can test it.